.github/workflows/ci_linux_arm64_clang.yml

```diff
@@ -71,3 +67,11 @@ jobs:
         run: ./build_tools/cmake/ctest_all.sh "${BUILD_DIR}"
       - name: Test iree-dialects
         run: ./build_tools/cmake/test_iree_dialects.sh "${BUILD_DIR}"
+
+      - name: Post to Discord on Failure
+        uses: sarisia/actions-status-discord@ce8cc68e4e626000136b3c702d049a154243e490 # v1.14.7
+        if: failure() && github.ref_name == 'main' && github.repository_owner == 'iree-org'
+        with:
+          webhook: ${{ secrets.DISCORD_WEBHOOK }}
+          description: "The ${{ github.workflow }} workflow failed"
+          url: "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}/attempts/${{ github.run_attempt }}"
```

Comment:
I considered using `workflow_summary.yml` in these places too but decided against that since these workflows only have a single job to "summarize".

Researched "composite actions" as a potential solution to reducing code duplication but the docs are confusing :P

* https://docs.github.com/en/actions/sharing-automations/avoiding-duplication
* https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-composite-action